application/src/main/java/org/thingsboard/server/service/ws/DefaultWebSocketService.java

```diff
@@ -420,6 +444,119 @@ private boolean processSubscription(WebSocketSessionRef sessionRef, Subscription
         return true;
     }
 
+    private void handleWsRpcCmd(WebSocketSessionRef sessionRef, RpcCmd cmd) {
```

Comment:
@lsyer 
I assume this is mainly a copy-paste from AbstractRpcController. This needs to be refactored. 
I guess the best solution would be to move the code from AbstractRpcController to TbRpcService, and reuse it here. 
For now, we will merge this to another feature branch and do the refactoring there. If you wish, you may do it yourself or wait for us to do it :)
Thanks for the contribution!

Reply:
You're too smart, indeed. Thank you.